library/packages/src/lib/packages/file_conflict_callbacks.rb

```diff
@@ -0,0 +1,185 @@
+
+      # Handle the file conflict detection start callback.
+      def start
+        log.info "Starting the file conflict check..."
+        # TRANSLATORS: progress bar label
+        label = _("Checking file conflicts...")
+
+        if Yast::Mode.commandline
+          Yast::CommandLine.PrintVerbose(label)
+        elsif pkg_installation?
+          # package slideshow with progress already present
+          Yast::UI.ChangeWidget(Id(PKG_INSTALL_WIDGET), :Value, 0)
+          Yast::UI.ChangeWidget(Id(PKG_INSTALL_WIDGET), :Label, label)
+        else
+          # TRANSLATORS: help text for the file conflict detection progress
+          help = _("<p>Detecting the file conflicts is in progress.</p>")
+          # Use the same label for the window title and the progressbar label
+          Yast::Progress.Simple(label, label, 100, help)
+        end
+      end
+
+      # Handle the file conflict detection progress callback.
+      # @param [Fixnum] progress progress in percents
+      # @return [Boolean] true = continue, false = abort
+      def progress(progress)
+        log.debug "File conflict progress: #{progress}%"
+
+        if Yast::Mode.commandline
+          Yast::CommandLine.PrintVerboseNoCR("#{Yast::PackageCallbacksClass::CLEAR_PROGRESS_TEXT}#{progress}%")
+        elsif pkg_installation?
+          Yast::UI.ChangeWidget(Id(PKG_INSTALL_WIDGET), :Value, progress)
+        else
+          Yast::Progress.Step(progress)
+        end
+
+        ui = Yast::UI.PollInput unless Yast::Mode.commandline
+        log.info "User input in file conflict progress (#{progress}%): #{ui}" if ui
+
+        ui != :abort && ui != :cancel
+      end
+
+      # Handle the file conflict detection result callback.
+      # Ask to user whether to continue. In the AutoYaST mode an error is reported
+      # but the installation will continue ignoring the confliucts.
+      # @param excluded_packages [Array<String>] packages ignored in the check
+      #   (e.g. not available for check in the download-as-needed mode)
+      # @param conflicts [Array<String>] list of translated descriptions of
+      #   the detected file conflicts
+      # @return [Boolean] true = continue, false = abort
+      def report(excluded_packages, conflicts)
+        log.info "Excluded #{excluded_packages.size} packages in file conflict check"
+        log.debug "Excluded packages: #{excluded_packages.inspect}"
+        log.info "Found #{conflicts.size} conflicts: #{conflicts.join("\n\n")}"
+
+        # just continue installing packages if there is no conflict
+        return true if conflicts.empty?
+
+        # don't ask in autoyast or command line mode, just report/log the issues and continue
+        if Yast::Mode.auto || Yast::Mode.commandline
+          # TRANSLATORS: An error message, %s is the actual list of detected conflicts
+          Yast::Report.Error(_("File conflicts detected, these conflicting files will " \
+            "be overwritten:\n\n%s") % conflicts.join("\n"))
```

Comment:
`join "\n\n"` also here? IIUC a single conflict is a multiline description, like 
```
File /usr/share/conflict
  from install of
     pkgconflict1-0.0.1-1.1.noarch (testing-conflict)
  conflicts with file from install of
     pkgconflict2-0.0.1-2.1.noarch (testing-conflict)
```